src/movies.js

```diff
@@ -1,22 +1,55 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const mappedArray = movies.map(function (singleMovie){
+    return singleMovie.director;
+   })
+   return mappedArray;
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies){
+  const movieDirector = movies.filter( (directorName) => {
+    return directorName.director === 'Steven Spielberg';
+  })
+  for (let i=0; movieDirector === 'Steven Spielberg'; i++){
+    const directorDrama = movieDirector.filter( (dramaMovie) => {
+      return dramaMovie.genre === 'Drama';
+    })
+  }
+  return directorDrama;
 
+}
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  let movieScores = movies.reduce(function (accValue, currentValue){
+    return accValue.score + currentValue.score
+  })
+  return movieScores
 
+}
 // Iteration 4: Drama movies - Get the average of Drama Movies
 function dramaMoviesScore() {}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear (movies) {
+  const newArray = movies.map(function (movieYear){
+    return movieYear.year
+   })
+  return newArray.sort();
+}
```

Comment:
1. Pay attention that task asks to return a new sorted array. We mentioned yesterday that sort() method changes original array, so before you use it you need to make a copy of original array.

couple of options to clone an array:
-> spread operator:
const newArray = [...originalArray]
-> slice():
const newArray = originalArray.slice()

2. The task asks you to sort the array, and return new sorted array of movie objects. Does map method really work here? Maybe we can use sort method instead of map... Check sort method syntax in student portal lecture ''Arrays Sort & Reverse" 

Sort method iterates through your arrays and compares two values:

function compare(a, b) {
    if (a is more than b) {
       return 1;
    }
    if (a is less than b) {
       return -1;
    }
}